addons/mail/models/models.py

```diff
@@ -46,6 +46,8 @@ def _mail_track(self, tracked_fields, initial):
                     tracking_sequence = 100
                 tracking = self.env['mail.tracking.value'].create_tracking_values(initial_value, new_value, col_name, col_info, tracking_sequence, self._name)
                 if tracking:
+                    if tracking['field_type'] == 'monetary':
+                        tracking['currency_id'] = self[col_info['currency_field']].id if 'currency_field' in col_info else self.company_id.currency_id.id
```

Comment:
Maybe `self.get('currency_field', self.company_id.currency_id).id would be a bit more elegant :) `